specs/carousel/carousel.md

```diff
@@ -0,0 +1,206 @@
+
+### Features
+
+- **Next and Previous Controls:** Simple buttons [aka flippers] that allow for control of displaying the next and previous slides.
+
+- **Slide Picker:** A control or group of controls that allows the specific choice of slide to display, labeled as the tab list.
+
+- **Autoplay:** Allows the carousel to begin rotating through slides automatically after the carousel loads. Autoplay can be toggled on/off with a button, and automatically stops when any element in the carousel receives keyboard focus or is hovered over.
+
+- **Looping:** Allows the user to navigate from the first slide to the last if the previous button is activated. If on the last slide the user can navigate from the last slide to the first when the next button is activated.
+
+- **ActiveSlideElement:** Provides a reference to the currently active slide from the `change` event.
+
+- **ActiveSlideId:** Provides a way to set the active slide.
+
+
+### Prior Art/Examples
+- [FAST-DNA (React)](https://explore.fast.design/components/carousel)
+- [W3C](https://w3c.github.io/aria-practices/examples/carousel/carousel-1.html)
+
+---
+
+## Design
+
+### API
+
+*Component name:*
+- `fast-carousel`
+
+*Attributes & Properties:*
+- `autoplay` - boolean value to toggle if the carousel plays on load or not (defaults to false)
+- `autoplay-interval` - a number representing the autoplay interval in milliseconds
+- `loop` - boolean value to toggle if the carousel loops slides at the beginning or end (defaults to true)
+- `aria-labelledby` - optional based on implementation**
+- `aria-label` - optional, based on implementation**
+- `paused` - boolean whether the rotation is paused or not
+- `activeslideid` - string
+
+** See the [W3C Specification](https://w3c.github.io/aria-practices/#wai-aria-roles-states-and-properties-4) for requirements and details.
+
+
+*Slots:*
+- `default` - use default slot for slide content
+- `play-toggle` - the control to toggle if the slides rotate or not
+- `next-button` - the control used to rotate to the next slide
+- `previous-button` - the control used to rotate to the previous slide
+- `tab-list` - the control to select a certain slide
+
+*Events*
+- `change` - callback fired when the slide changes. This updates the `activeslideid` and passes a reference to the active slide. The play/pause state and `activeslideid` can then be read by the author from the reference element, e.g. event.target.paused, event.target.activeslideid
+
+### Anatomy and Appearance
+**Structure:**
+
+Notes:
+- Since the host element has the handleClick(), it would use event delegation to check what element was clicked on. Example: If the target was an element with a `[slot="next-button"]` it would trigger the controller method for nextSlide(). This way each element passed in via slot would not needs it's own onClick.
+
+```html
+<!-- shadow dom -->
+<div
+    class="carousel"
+    aria-roledescription="carousel"
+    tabindex="-1"
+    @click="${x => x.handleClick()}"
```

Comment:
If this is the shadow dom we wouldn't see the click handler here.

Reply:
I think this will be solved by attaching the click handler to the host itself. That should let it see the clicks from the light dom child nodes. Good catch @eljefe223 !